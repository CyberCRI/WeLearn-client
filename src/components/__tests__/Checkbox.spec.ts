import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';
import CheckboxCompnentVue from '../CheckboxCompnent.vue';

describe('CheckboxCompnentVue', () => {
  it('renders properly', () => {
    const wrapper = mount(CheckboxCompnentVue, {
      propsData: {
        label: 'i am a label',
        value: 'i am a value',
        id: 'i am an id',
        name: 'i am a name',
        modelValue: ['i am a modelValue'],
        checked: true
      }
    });
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).toContain('i am a label');
  });
});
