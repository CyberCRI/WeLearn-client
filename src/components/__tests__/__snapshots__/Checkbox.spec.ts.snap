// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CheckboxCompnentVue > renders properly 1`] = `
<label
  aria-label="i am a value"
  class="container"
  data-v-7932c334=""
>
  <div
    class="checkbox-label"
    data-v-7932c334=""
  >
    <p
      data-v-7932c334=""
    >
      i am a label
    </p>
    
    
  </div>
  <input
    checked=""
    data-v-7932c334=""
    id="i am an id"
    type="checkbox"
    value="i am a value"
  />
  <span
    class="checkmark"
    data-v-7932c334=""
  />
</label>
`;
